src/test/tests.ts

```diff
@@ -896,7 +893,15 @@ describe('Server', function () {
       //do nothing
     });
 
-    client.unsubscribe(subId);
+    setTimeout(() => {
+      // TODO
+      // Hagaico I had the need to add this timeout here
```

Comment:
I think the major difference is that before isn't verified if the operation is into the operations array and we just call subscriptions manager unsubscribe and the onUnsubscribe callback

```javascript
private unsubscribe(connection: WebSocket, handleId: number) {
    this.subscriptionManager.unsubscribe(handleId);

    if (this.onUnsubscribe) {
      this.onUnsubscribe(connection);
    }
  }
```

Now we are verifying if the operation is registered on the server, so I bet that timeout is needed to allow the handshake, the message with subscribe being received and registered by the server. Do u agree @DxCx ?

```javascript
private unsubscribe(connectionContext: ConnectionContext, opId: string) {
    if (connectionContext.operations && connectionContext.operations[opId]) {
      connectionContext.operations[opId].unsubscribe();
      delete connectionContext.operations[opId];

      if (this.onOperationComplete) {
        this.onOperationComplete(connectionContext.socket);
      }
    }
  }
```